wp/headless-wp/includes/classes/Integrations/Gutenberg.php

```diff
@@ -22,6 +22,94 @@ class Gutenberg {
 	 */
 	public function register() {
 		add_filter( 'render_block', [ $this, 'render_block' ], 10, 3 );
+		add_filter( 'render_block_core/image', [ $this, 'ensure_image_has_dimensions' ], 9999, 2 );
+	}
+
+	/**
+	 * Get the image ID by URL
+	 *
+	 * @param string $url The image URL
+	 *
+	 * @return int
+	 */
+	protected function get_image_by_url( $url ) {
+		if ( function_exists( '\wpcom_vip_attachment_url_to_postid' ) ) {
+			return \wpcom_vip_attachment_url_to_postid( $url );
+		}
+
+		$cache_key = sprintf( 'get_image_by_%s', md5( $url ) );
+		$url       = esc_url_raw( $url );
+		$id        = wp_cache_get( $cache_key, 'headstartwp', false );
+
+		if ( false === $id ) {
+			$id = attachment_url_to_postid( $url );
+
+			/**
+			 * If no ID was found, maybe we're dealing with a scaled big image. So, let's try that.
+			 *
+			 * @see https://core.trac.wordpress.org/ticket/51058
+			 */
+			if ( empty( $id ) ) {
+				$path_parts = pathinfo( $url );
+
+				if ( isset( $path_parts['dirname'], $path_parts['filename'], $path_parts['extension'] ) ) {
+					$scaled_url = trailingslashit( $path_parts['dirname'] ) . $path_parts['filename'] . '-scaled.' . $path_parts['extension'];
+					$id         = attachment_url_to_postid( $scaled_url );
+				}
+			}
+
+			wp_cache_set( $cache_key, $id, 'headstartwp', 3 * HOUR_IN_SECONDS );
+		}
+
+		return $id;
+	}
+
+	/**
+	 * Ensure that images have dimensions set
+	 *
+	 * @param string $block_content the html for the block
+	 * @param array  $block the block's schema
+	 *
+	 * @return string
+	 */
+	public function ensure_image_has_dimensions( $block_content, $block ) {
+		/**
+		 * Filter whether to bypass adding dimensions to images
+		 *
+		 * @param bool   $bypass          Whether to bypass adding dimensions, defaults to false
+		 * @param string $block_content   The block content
+		 * @param array  $block          The block schema
+		 */
+		if ( ! apply_filters( 'tenup_headless_wp_ensure_image_dimensions', false, $block_content, $block ) ) {
+			return $block_content;
+		}
+
+		$doc = new \WP_HTML_Tag_Processor( $block_content );
```

Comment:
does this need also to have a check of existing class and value of this filter https://github.com/10up/headstartwp/blob/develop/wp/headless-wp/includes/classes/Integrations/Gutenberg.php#L262 ?

Reply:
Not really, that filter is intenteded only for the block parsing per se and it only exists because this API was introduced after the first release of the plugin and I added it to avoid breaking changes as there are some subtle changes in parsing.